maestro-client/src/main/java/maestro/drivers/IOSDriver.kt

```diff
@@ -181,7 +178,10 @@ class IOSDriver(
     }
 
     override fun contentDescriptor(): TreeNode {
-        return viewHierarchy()
+        return when (val contentDescriptorResult = iosDevice.contentDescriptor()) {
+            is Ok -> mapHierarchy(contentDescriptorResult.value)
+            is Err -> TreeNode()
+        }
```

Comment:
Why are we using Result here, instead of handling exceptions?

Reply:
In case of unknown errors we just return empty hierarchy. This is in parity with production

Reply:
I mean, why do we use Result in IOSDevice for each methods return type, instead of using exception throwing?
Inside IOSDriver all Result values are then `.expect {}`'ed anyway..

Reply:
No action required here though, we can have a look at this later.